lms/services/blackboard_api/client.py

```diff
@@ -113,4 +114,35 @@ def group_set_groups(self, course_id, group_set_id):
             "GET",
             f"/learn/api/public/v2/courses/uuid:{course_id}/groups/sets/{group_set_id}/groups",
         )
-        return BlackboardListGroupSetGroups(response).parse()
+        return BlackboardListGroupsSchema(response).parse()
+
+    def course_groups(self, course_id, group_category_id=None, user_id=None):
+        response = self._api.request(
+            "GET",
+            f"/learn/api/public/v2/courses/uuid:{course_id}/groups",
+        )
+        groups = BlackboardListGroupsSchema(response).parse()
+
+        if group_category_id:
+            groups = [
+                group
+                for group in self.course_groups(course_id)
+                if group["groupSetId"] == group_category_id
+            ]
+
+        if user_id:
+            groups_by_id = {group["id"]: group for group in groups}
+            responses = self._api.async_requests(
+                "GET",
+                [
+                    f"/learn/api/public/v2/courses/uuid:{course_id}/groups/{group['id']}"
+                    for group in groups
+                ],
+            )
+            groups = [
+                groups_by_id[str(response.url).split("/")[-1]]
```

Comment:
response.url has the full URL (including schema, domain...) instead of that the path we use here.